### ISSUE TITLE
res_imedcy not working on first run

### ISSUE THREAD

**Issue by Bullenj01** [terraform-providers/terraform-provider-aci#104]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
13.1

### APIC version and APIC Platform

4.2

### Affected Resource(s)

aci_epg_to_domain

### Terraform Configuration Files

```hcl
resource "aci_application_epg" "vmware-mgmt" {
  application_profile_dn = aci_application_profile.vmware-mgmt.id
  name                   = "vmware-mgmt"
  relation_fv_rs_dom_att = [
    data.aci_vmm_domain.knxvcenter01t.id,
    data.aci_physical_domain.legacy.id,
  ]
  relation_fv_rs_bd      = aci_bridge_domain.vmware-mgmt.id
}
resource "aci_epg_to_domain" "vmware-mgmt_knxvcenter01t" {
  application_epg_dn = aci_application_epg.vmware-mgmt.id
  tdn = data.aci_vmm_domain.knxvcenter01t.id
  res_imedcy = "pre-provision" #This is needed on vmkernal mgmt ports, chicken egg kinda thing
```

### Expected Behavior

res_imedcy should be set correctly on first run

### Actual Behavior

res_imedcy requires a second run to actually apply

### Steps to Reproduce

1. `terraform apply`

1st Run

Terraform will perform the following actions:

  aci_application_epg.vmware-mgmt will be updated in-place
  ~ resource "aci_application_epg" "vmware-mgmt" {
        application_profile_dn = "uni/tn-prod/ap-192.131.32.0_24"
        flood_on_encap         = "disabled"
        has_mcast_source       = "no"
        id                     = "uni/tn-prod/ap-192.131.32.0_24/epg-vmware-mgmt"
        is_attr_based_epg      = "no"
        match_t                = "AtleastOne"
        name                   = "vmware-mgmt"
        pc_enf_pref            = "unenforced"
        pref_gr_memb           = "exclude"
        prio                   = "unspecified"
        relation_fv_rs_bd      = "uni/tn-prod/BD-vmware-mgmt"
      ~ relation_fv_rs_dom_att = [
            "uni/phys-legacy",
          + "uni/vmmp-VMware/dom-knxvcenter01t",
        ]
        shutdown               = "no"
    }

  aci_epg_to_domain.vmware-mgmt_knxvcenter01t will be created
  + resource "aci_epg_to_domain" "vmware-mgmt_knxvcenter01t" {
      + annotation            = (known after apply)
      + application_epg_dn    = "uni/tn-prod/ap-192.131.32.0_24/epg-vmware-mgmt"
      + binding_type          = (known after apply)
      + class_pref            = (known after apply)
      + delimiter             = (known after apply)
      + description           = (known after apply)
      + encap                 = (known after apply)
      + encap_mode            = (known after apply)
      + epg_cos               = (known after apply)
      + epg_cos_pref          = (known after apply)
      + id                    = (known after apply)
      + instr_imedcy          = (known after apply)
      + lag_policy_name       = (known after apply)
      + netflow_dir           = (known after apply)
      + netflow_pref          = (known after apply)
      + num_ports             = (known after apply)
      + port_allocation       = (known after apply)
      + primary_encap         = (known after apply)
      + primary_encap_inner   = (known after apply)
      + res_imedcy            = "pre-provision"
      + secondary_encap_inner = (known after apply)
      + switching_mode        = (known after apply)
      + tdn                   = "uni/vmmp-VMware/dom-knxvcenter01t"
      + vmm_allow_promiscuous = (known after apply)
      + vmm_forged_transmits  = (known after apply)
      + vmm_id                = (known after apply)
      + vmm_mac_changes       = (known after apply)
    }

2nd run after apply above

  aci_epg_to_domain.vmware-mgmt_knxvcenter01t will be updated in-place
  ~ resource "aci_epg_to_domain" "vmware-mgmt_knxvcenter01t" {
        application_epg_dn    = "uni/tn-prod/ap-192.131.32.0_24/epg-vmware-mgmt"
        binding_type          = "none"
        class_pref            = "encap"
        encap                 = "unknown"
        encap_mode            = "auto"
        epg_cos               = "Cos0"
        epg_cos_pref          = "disabled"
        id                    = "uni/tn-prod/ap-192.131.32.0_24/epg-vmware-mgmt/rsdomAtt-[uni/vmmp-VMware/dom-knxvcenter01t]"
        instr_imedcy          = "lazy"
        netflow_dir           = "both"
        netflow_pref          = "disabled"
        num_ports             = "0"
        port_allocation       = "none"
        primary_encap         = "unknown"
        primary_encap_inner   = "unknown"
      ~ res_imedcy            = "lazy" -> "pre-provision"
        secondary_encap_inner = "unknown"
        switching_mode        = "native"
        tdn                   = "uni/vmmp-VMware/dom-knxvcenter01t"
        vmm_allow_promiscuous = "reject"
        vmm_forged_transmits  = "reject"
        vmm_id                = "uni/tn-prod/ap-192.131.32.0_24/epg-vmware-mgmt/rsdomAtt-[uni/vmmp-VMware/dom-knxvcenter01t]/sec"
        vmm_mac_changes       = "reject"
--->

* #0000

**Comment by Bullenj01** [comment]
its possible, the problem was because I was using relation_fv_rs_dom_att under the aci_application_epg resource then following it up with a aci_epg_to_domain resource.  When I remove the relationship out of the aci_application_epg resource and just use the dedicated resource for this I don't have this problem anymore

**Comment by Bullenj01** [comment]
spoke to soon.  without the relation_fv_rs_dom_att under the epg you end up in a terraform apply loop where the first run it creates the domain from the aci_epg_to_domain resource but if you run terraform apply again you get this:
```
 ~ resource "aci_application_epg" "credit_dmz" {
        application_profile_dn = "uni/tn-dev/ap-10.5.241.0_24"
        flood_on_encap         = "disabled"
        has_mcast_source       = "no"
        id                     = "uni/tn-dev/ap-10.5.241.0_24/epg-credit_dmz"
        is_attr_based_epg      = "no"
        match_t                = "AtleastOne"
        name                   = "credit_dmz"
        pc_enf_pref            = "unenforced"
        pref_gr_memb           = "include"
        prio                   = "unspecified"
        relation_fv_rs_bd      = "uni/tn-dev/BD-credit_dmz"
      ~ relation_fv_rs_dom_att = [
          - "uni/vmmp-VMware/dom-vmware_aci",
        ]
        shutdown               = "no"
    }
```

**Comment by Bullenj01** [comment]
@nkatarmal-crest just looked at your pull request and see that you are removing that relation as a choice under the epg which is just what I was about to suggest...just a few days late.  I should be good after the next release.  thanks!

**Comment by nkatarmal-crest** [comment]
@Bullenj01 I have fixed this issue in v0.4.0 of the provider. Feel free to reopen the issue if you are still facing it.
